tests/functional/u/unnecessary/unnecessary_dict_index_lookup.py

```diff
@@ -82,3 +82,15 @@ class Foo:
     if 'V' in d[k]:  # [unnecessary-dict-index-lookup]
         d[k] = "value"
         print(d[k])  # This is fine
+
+# Test false positive described in #4716
+# Should not be emitted for del
+# (https://github.com/PyCQA/pylint/issues/4716)
+d = {}
+for key, val in d.items():
+    del d[key]
+    break
+
+for item in d.items():
+    del d[item[0]]
+    break
```

Comment:
```suggestion
    break

outer_dict = {"inner_dict": {}}
for key, val in outer_dict.items():
    for key_two, val_two in val.items():
        del outer_dict[key][key_two]  # [unnecessary-dict-index-lookup]
        break
```

Additional test to see whether we're not ignoring all subscripts on a `del` statement.

Reply:
done! it works!